### PR TITLE
add fullscreen support

### PR DESCRIPTION
**Pull request by xintongxia**
https://github.com/uber/react-map-gl/issues/668

### CODE REVIEW

File: src/components/fullscreen-control.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+const propTypes = Object.assign({}, BaseControl.propTypes, {
+  // Custom className
+  className: PropTypes.string,
+  /* eslint-disable max-len */
+  // `container` is the [compatible DOM element](https://developer.mozilla.org/en-US/docs/Web/API/Element/requestFullScreen#Compatible_elements)
+  // which should be made full screen. By default, the map container element will be made full screen.
+  /* eslint-enable max-len */
+  container: PropTypes.object,
+  // show fullscreen button
+  showFullscreen: PropTypes.bool
```
Comment (Pessimistress) [inline src/components/fullscreen-control.js]: Why is this needed?

File: src/utils/fullscreen.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
+import {document} from './globals';
```
Comment (Pessimistress) [inline src/utils/fullscreen.js]: I prefer not having to include this in our library if we could just invoke Mapbox's native FullScreenControl.

File: docs/components/fullscreen-control.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
+# Fullscreen Control
```
Comment (Pessimistress) [inline docs/components/fullscreen-control.md]: Add to website

File: src/components/fullscreen-control.js [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+    this._mapboxFullscreenControl = null;
+
+    this.state = {
+      isFullscreen: false
+    };
+  }
+
+  componentDidMount() {
+    const container = this.props.container || this._context.mapContainer;
```
Comment (Pessimistress) [inline src/components/fullscreen-control.js]: `StaticMap` can be used standalone. `context.mapContainer` will be empty in that case.

Comment (xintongxia) [reply src/components/fullscreen-control.js]: My understanding is `StaticMap` should also support `FullscreenControl`. Then I need find a way to get the `MapContainer`(container also include the controls) in standalone `StaticMap` case.

File: src/components/static-map.js [hunk 5]
```diff
@@ -166,6 +166,7 @@ export default class StaticMap extends PureComponent<StaticMapProps, State> {
   _mapbox : any = null;
   _map : any = null;
   _mapboxMapRef: { current: null | HTMLDivElement } = createRef();
+  _mapContainer: { current: null | HTMLDivElement } = createRef();
```
Comment (Pessimistress) [inline src/components/static-map.js]: Nit: call it `_mapContainerRef`

**Comment by macobo** [comment]
Woop! Really happy to have this, thanks @xintongxia and @Pessimistress, looking forward to the next release. :)
